### ISSUE TITLE
Support for Auth redirections and Scope violations

### ISSUE THREAD

**Issue by tchoutri** [milmazz/hunter#9]
Hi @milmazz :)
I've noticed that the process calling `Hunter.log_in` crashes systematically when the server returns a `302 Redirect` with an HTML response body (which happens when [the login fails](https://github.com/tootsuite/documentation/blob/master/Using-the-API/Testing-with-cURL.md))
The same behaviour occurs when I try to use `create_status` with access tokens for a `read only` application.

Since it would mean some significant changes in the HTTP Client's internal design, I wanted your opinion on whether we should handle those cases or leave it to library users to manage them.
Thanks :)

**Comment by milmazz** [comment]
@tchoutri I think that you should receive a `Hunter.Error` exception in those cases, which I think it's fine, then, the library users can manage or rescue that case with something like this:

```elixir
try do
  Hunter.log_in(app, username, passwd)
rescue ex in [Hunter.Error] ->
  ...
end
```

If you think that the current error message can be improved what we can do is include a `try` and `rescue` in the `log_in` function, improve the error message including the `redirect_uri` and then re-raise the error as an `ArgumentError`.

Please let me know what do you think.

**Comment by tchoutri** [comment]
It actually throws a MatchError (I don't have my laptop with me at the moment), because it can't match the HTML response, or the http code. I think it happens when Poison tries to convert the reponse to a Struct, so in the `transform` function.
Thanks for your answers :)

**Comment by milmazz** [comment]
@tchoutri I just checked and the current behavior of `Hunter.log_in/4` seems to be OK:

```elixir
iex(1)> app = Hunter.load_credentials("hunter")
%Hunter.Application{client_id: "1234567890", client_secret: "1234567890",
 id: 1234}
iex(2)> conn = Hunter.log_in(app, "jdoe@example.com", "your_password")
** (Hunter.Error) "Invalid Email or password."
    (hunter) lib/hunter/request.ex:16: Hunter.Request.request!/5
    (hunter) lib/hunter/api/http_client.ex:301: Hunter.Api.HTTPClient.log_in/4
```

Notice that the `Hunter.Error` seems to be pretty clear in this case. But, please let me know what do you think.

BTW, your screenshot seems to be another error, do you want to send a PR including the fix?

**Comment by tchoutri** [comment]
Hi! Well, the JSON body that is returned says that my action is "outside the authorized scopes". The funny thing is that I tried to register other applications with the whole "read", "write", "follow", on two different servers but I couldn't make it work. I'll ask friends of mine to test `create_status`.

For the 2nd error, I realized yesterday that the "username" parameter was actually an email address, so I'm at fault here. =)

I could try to fix the MatchError, yeah. I'll try some ideas later today.
